policy_iteration.py

```diff
@@ -0,0 +1,89 @@
+# Author: Grace Calkins
+# Date: 2/15/2023
+# For AE 598RL, UIUC Spring 2023
+
+import numpy as np
+
+
+def poli_iter(env, gamma = 0.95):
```

Comment:
I think you've done a very good job in writing the code! The code is well-commented and understandable. One suggestion that I personally think can be useful to make you code more modular and easier to follow (which you might not like and it's very subjective :) ) is to break down the policy iteration algorithm into the main two steps: 1) policy evaluation, and 2) policy improvement and write a separate function for each step separately, and them combine the two functions into a "policy_iter" function. This way, your code will be more modular and the steps that you need to implement the PI algorithm will be only a single line of code inside the main "policy_iter" function, and the reader can understand the main steps easier.

Reply:
Hi Mahshid, I implemented your suggestion of breaking policy iteration and policy improvement up into 2 methods and calling them within the main policy iteration function. :)